aya/src/maps/mod.rs

```diff
@@ -292,18 +292,65 @@ impl Map {
         }
         let map_path = path.as_ref().join(name);
         let path_string = CString::new(map_path.to_str().unwrap()).unwrap();
-        let fd =
-            bpf_get_object(&path_string).map_err(|(code, io_error)| MapError::SyscallError {
-                call: "BPF_OBJ_GET".to_string(),
-                code,
-                io_error,
-            })? as RawFd;
+        let fd = bpf_get_object(&path_string).map_err(|(_, io_error)| MapError::SyscallError {
+            call: "BPF_OBJ_GET".to_string(),
+            io_error,
+        })? as RawFd;
 
         self.fd = Some(fd);
 
         Ok(fd)
     }
 
+    /// Loads a map from a pinned path in bpffs.
+    pub fn from_pinned<P: AsRef<Path>>(path: P) -> Result<Map, MapError> {
```

Comment:
This is from_pinned but in another PR we're adding from_path. We should agree on one.